### PR TITLE
Add match! to structure

### LINKED ISSUE #5456: match! breaks structure guidelines and collapsible nodes
Structure guidelines and collapsible nodes below a `match!` are missing.

#### Repro steps

```fsharp
let blah = async {
    match! async { return Some 1 } with
    | Some _ -> ()
    | None ->
        if true then
            if true then
                if true then
                    () }
```

#### Expected behavior

Structure guidelines and collapsible nodes are present.

#### Actual behavior

untitled

#### Known workarounds

None.

#### Related information 

Provide any related information 

VS 2017.8 Preview 5

### PR DESCRIPTION
**Pull request by cartermp**
Fixes #5456 

image

### CODE REVIEW

File: src/fsharp/service/ServiceStructure.fs [hunk 1]
```diff
@@ -259,7 +260,7 @@ module Structure =
             | SynExpr.LetOrUse (_,_,bindings, body, _) ->
                 parseBindings bindings
                 parseExpr body
-            | SynExpr.Match (seqPointAtBinding,_expr,clauses,_,r) ->
+            | SynExpr.Match (seqPointAtBinding,_expr,clauses,_,r) | SynExpr.MatchBang (seqPointAtBinding, _expr, clauses, _, r) ->
```
Comment (vasily-kirichenko) [inline src/fsharp/service/ServiceStructure.fs]: I’m sure there no such formatting in the whole repository.

File: tests/service/StructureTests.fs [hunk 2]
```diff
@@ -301,6 +301,25 @@ match None with     // 2
...
+    """
+match! None with     // 2
+| Some _ ->          // 3
+    ()               // 4
+| None ->            // 5
+    match None with  // 6
+    | Some _ -> ()   // 7
+    | None ->        // 8
+        let x = ()   // 9
+        ()           // 10
```
Comment (vasily-kirichenko) [inline tests/service/StructureTests.fs]: The example code is not realistical.

File: src/fsharp/service/ServiceStructure.fs (hunk 1 above)
Comment (cartermp) [reply src/fsharp/service/ServiceStructure.fs]: There is: https://github.com/Microsoft/visualfsharp/blob/master/src/fsharp/service/ServiceUntypedParse.fs#L872

File: tests/service/StructureTests.fs (hunk 2 above)
Comment (cartermp) [reply tests/service/StructureTests.fs]: This is just like the other tests. Do you have a suggested improvement?

Comment (vasily-kirichenko) [reply tests/service/StructureTests.fs]:
```fsharp
maybe {
    match! Some None with     
    | Some _ ->          
        ()               
    | None ->            
        match None with  
        | Some _ -> ()   
        | None ->        
            let x = ()   
            ()
}                       
```

File: src/fsharp/service/ServiceStructure.fs (hunk 1 above)
Comment (KevinRansom) [reply src/fsharp/service/ServiceStructure.fs]: @cartermp,

From a readability perspective, I think we should prefer the fall through match,  I agree with @vasily-Kirichenko.

It is a certainty that we can using this repo point to an example of every type of formatting style that could possibly exist.  For sure fall through matches with the separate cases per line is the most common, and I think readable format.

Kevin

Comment (KevinRansom) [reply src/fsharp/service/ServiceStructure.fs]: @cartermp,
From a readability perspective, I think we should prefer the fall through match, I agree with @vasily-kirichenko.
It is a certainty that we can using this repo point to an example of every type of formatting style that could possibly exist. For sure fall through matches with the separate cases per line is the most common, and I think readable format.
Kevin

**Pull request merged**